packages/noya-renderer/src/components/DistanceLabelAndPath.tsx

```diff
@@ -0,0 +1,323 @@
+      { x: selectedBounds.maxX, y: selectedBounds.midY },
+      {
+        x: rightEdgeOfSelectedBounds,
+        y: selectedBounds.midY,
+      },
+    );
+
+    distanceMeasurements.push({
+      bounds: {
+        x: selectedBounds.maxX + itemDistance / 2,
+        y: selectedBounds.midY,
+      },
+      distance: itemDistance,
+    });
+  }
+
+  const startX =
+    highlightedBounds.midX < selectedBounds.midX
+      ? highlightedBounds.maxX
+      : highlightedBounds.minX;
+
+  const endX =
+    highlightedBounds.midX < selectedBounds.midX
+      ? selectedBounds.maxX
+      : selectedBounds.minX;
+
+  let topEdgeOfSelectedBounds =
+    highlightedBounds.maxY < selectedBounds.minY
+      ? highlightedBounds.maxY
+      : highlightedBounds.maxY < selectedBounds.maxY
+      ? highlightedBounds.minY
+      : undefined;
+
+  if (topEdgeOfSelectedBounds !== undefined) {
+    implicitLines.push([
+      {
+        x: startX,
+        y: topEdgeOfSelectedBounds,
+      },
+      {
+        x: endX,
+        y: topEdgeOfSelectedBounds,
+      },
+    ]);
+
+    explicitLines.push([
+      { x: selectedBounds.midX, y: selectedBounds.minY },
+      {
+        x: selectedBounds.midX,
+        y: topEdgeOfSelectedBounds,
+      },
+    ]);
+
+    const itemDistance = distance(
+      { x: selectedBounds.midX, y: selectedBounds.minY },
+      {
+        x: selectedBounds.midX,
+        y: topEdgeOfSelectedBounds,
+      },
+    );
+
+    distanceMeasurements.push({
+      bounds: {
+        x: selectedBounds.midX,
+        y: selectedBounds.minY - itemDistance / 2,
+      },
+      distance: itemDistance,
+    });
+  }
+
+  let bottomEdgeOfSelectedBounds =
+    highlightedBounds.minY > selectedBounds.maxY
+      ? highlightedBounds.minY
+      : highlightedBounds.minY > selectedBounds.minY
+      ? highlightedBounds.maxY
+      : undefined;
+
+  if (bottomEdgeOfSelectedBounds !== undefined) {
+    implicitLines.push([
+      { x: startX, y: bottomEdgeOfSelectedBounds },
+      {
+        x: endX,
+        y: bottomEdgeOfSelectedBounds,
+      },
+    ]);
+
+    explicitLines.push([
+      { x: selectedBounds.midX, y: selectedBounds.maxY },
+      {
+        x: selectedBounds.midX,
+        y: bottomEdgeOfSelectedBounds,
+      },
+    ]);
+
+    const itemDistance = distance(
+      { x: selectedBounds.midX, y: selectedBounds.maxY },
+      {
+        x: selectedBounds.midX,
+        y: bottomEdgeOfSelectedBounds,
+      },
+    );
+
+    distanceMeasurements.push({
+      bounds: {
+        x: selectedBounds.midX,
+        y: selectedBounds.maxY + itemDistance / 2,
+      },
+      distance: itemDistance,
+    });
+  }
+
+  const { CanvasKit } = useReactCanvasKit();
+  const paint = new CanvasKit.Paint();
+  paint.setColor(CanvasKit.Color4f(0.52, 0.248, 1.0));
+  paint.setPathEffect(CanvasKit.PathEffect.MakeDash([1, 2]));
+  paint.setStyle(CanvasKit.PaintStyle.Stroke);
+  paint.setStrokeWidth(1);
+
+  const explicitPaint = new CanvasKit.Paint();
```

Comment:
Can we use `usePaint` here?

Reply:
yes, renamed to` measureGuidePaint` and used `usePaint` hook